### DISCUSSION TITLE
Is there a option to restart a machine/system in safe mode using mesh-central

### DISCUSSION THREAD

**Discussion by a-imran99** [Ylianst/MeshCentral#4457] [Ideas]
Hi @Ylianst , we have the options to restart a system/machine in which the mesh agent is installed. (under Action button).
Do we have a feature to restart or reboot the systems in safe mode using the mesh agent. If restart in safe mode is achieved will mesh agent still run on the systems which has safe mode turned on?

**Comment by a-imran99** [comment]
@Ylianst , some suggestion would be more helpful :)
Thank you!

**Comment by si458** [comment]
im not 100% sure but i think if you ask it to restart into safe mode,
it will restart into the safe mode with networking, so you would still be able to access it afterwards,
but i guess you would have to test it and find out haha

**Reply by a-imran99** [reply]
hey @si458 simon, its great you replied, 

just now tried with a system in which mesh agent is installed and booted it up with safe mode . This system doesn't show up as powered in mesh-central.

Tried the same in safe-mode with network drivers, since only basic drivers are installed and more over its not configured, couldn't actually connect to the internet by any mean and the agent too is not running in services.

Is there any way to run the agent even in safe-mode with networking. @Ylianst 

Thanks @si458...

**Reply by si458** [reply]
i have a feeling then that the 'safe mode' option is designed for Intel AMT where people have physical control of the computers remotely then

**Reply by silversword411** [reply]
> Is there any way to run the agent even in safe-mode with networking. @Ylianst

From my past experience/tests, hardwired ethernet connections will work with safe-mode with networking. Wireless will almost never, especially when you don't have physical access because the WLAN Autoconfig (previously wireless zero config) service isn't deemed a required "safe-mode with networking" service.

**Comment by Ylianst** [comment]
I see #591 and #3805 on this topic. Bryan has indicated he would add support for this a long time ago. Looking at the code, there is a "safemode" command in the agent console.

![image](https://user-images.githubusercontent.com/1319013/187044847-eff28e6b-925e-43e8-9ade-06696031d259.png)

I have not tried it and don't know if it will do safe mode with networking. This is certainly a question for Bryan.

**Reply by a-imran99** [reply]
> I see #591 and #3805 on this topic. Bryan has indicated he would add support for this a long time ago. Looking at the code, there is a "safemode" command in the agent console.

Thanks for the reply @Ylianst , So I guess that safe mode is only for AMT devices and not works with devices with ordinary agent installed!. 

Even with safe-mode with networking, it still seems impossible to take a remote access to a machine which has only WiFi as said by @silversword411.

Is it still possible to take remote control of a system in which mesh-agent is installed, while I tried it shows the system is not booted up if it is booted up in safe-mode in Mesh-central.

Thanks @Ylianst, @krayon007 , @silversword411. :)

**Comment by krayon007** [comment]
Support for SafeMode with networking should already be there... The installer will set the registry keys to enable the meshagent to run, when you restart in safemode... However, this will depend on when you installed the agent. I'll have to check when the installer was modified to do this.  

In anycase, you can set the modes at anytime, with the console commands.
`SAFEMODE ON`
This will set the registry keys to allow the agent to run in safemode with networking. It will also set the BCD key, so that next boot will be safeboot with networking.
If you immediately follow with:
`SAFEMODE OFF`
This will clear the BCD keys, so that nextboot will be normal, but it will leave the agent configured so that it will function correctly in safemode.

**Comment by krayon007** [comment]
So if you do: `SAFEMODE ON`
followed by a power action to restart, that should restart the machine in safemode with networking.

**Comment by krayon007** [comment]
If you do:
`SAFEMODE STATUS`
That should tell you what the nextboot state is... It should be either Normal, or Safemode/Networking
